src/cell_group_factory.cpp

```diff
@@ -14,26 +16,29 @@ namespace mc {
 using gpu_fvm_cell = mc_cell_group<fvm::fvm_multicell<gpu::backend>>;
 using mc_fvm_cell = mc_cell_group<fvm::fvm_multicell<multicore::backend>>;
 
-cell_group_ptr cell_group_factory(
-        cell_kind kind,
-        cell_gid_type first_gid,
-        const std::vector<util::unique_any>& cell_descriptions,
-        backend_policy backend)
-{
-    switch (kind) {
+cell_group_ptr cell_group_factory(const recipe& rec, const group_description& group) {
+    // Make a list of all the cell descriptions to be forwarded
+    // to the appropriate cell_group constructor.
+    std::vector<util::unique_any> descriptions;
+    descriptions.reserve(group.gids.size());
+    for (auto gid: group.gids) {
+        descriptions.push_back(rec.get_cell_description(gid));
+    }
+
+    switch (group.kind) {
```

Comment:
If we're going to make good on our modularity promise when it comes to cell groups, we are going to have to find a solution for registering both new cell kinds and new cell group implementations for a cell kind at run- or link-time: the decompositor/scheduler will need to decide which implementation to allocate a cell group to, based on the implementation's estimated performance characteristics (and in the GPU case, hardware availability).

But while worth considering and planning for, I don't think it's something we're going to pursue in the initial version.